src/components/PeopleTable/PeopleTable.tsx

```diff
@@ -0,0 +1,101 @@
+import React, { useEffect, useState } from 'react';
+import { Person } from '../../types';
+import { PersonLink } from '../PersonLink';
+import classNames from 'classnames';
+import { useParams } from 'react-router-dom';
+
+type Props = {
+  people: Person[] | null;
+};
+
+export const PeopleTable: React.FC<Props> = ({ people }) => {
+  const { personId } = useParams();
+  const [activePersonName, setActivePersonName] = useState<string | null>(null);
+
+  const isPersonInTable = (personName: string) => {
+    return people?.find(({ name }) => name === personName);
+  };
+
+  const handleLinkForParent = (personName: string) => {
+    if (!people) {
+      return;
+    }
+
+    if (isPersonInTable(personName)) {
+      const indexOfPerson = people?.findIndex(
+        person => person.name === personName,
+      );
+      const person = people[indexOfPerson];
+      const { name, sex, slug } = person;
+
+      return (
+        <PersonLink
+          name={name}
+          sex={sex}
+          slug={slug}
+        />
+      );
```

Comment:
The `handleLinkForParent` function returns a JSX element, which is not a typical behavior for a function that is expected to return a string or a component. Consider renaming the function to reflect its purpose more accurately or refactoring the logic.